core/node/node_framework/src/resource/mod.rs

```diff
@@ -39,6 +39,14 @@ pub(crate) trait StoredResource: 'static + std::any::Any + Send + Sync {
     fn stored_resource_wired(&mut self);
 }
 
+impl fmt::Debug for dyn StoredResource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Resource")
+            .field("resource_id", &self.stored_resource_id())
+            .finish()
+    }
+}
+
```

Comment:
why we do not implement `fmt::Debug` for `Resource` trait?

Reply:
Do you mean why don't we have `impl fmt::Debug for dyn Resource`? Because `Resource` is not object safe (that's the reason why we need `StoredResource` in the first place).

Reply:
>Because Resource is not object safe

not sure I understand why it should not be debug then..

Reply:
Not sure if I understand your question. Could you elaborate more? Here we implement `Debug` for `dyn StoredResource`, so that we can use it when we store the object as a dyn trait.